core/state/snapshot/snapshot.go

```diff
@@ -186,7 +186,11 @@ func New(diskdb ethdb.KeyValueStore, triedb *trie.Database, cache int, root comm
 		defer snap.waitBuild()
 	}
 	// Attempt to load a previously persisted snapshot and rebuild one if failed
-	head, err := loadSnapshot(diskdb, triedb, cache, root, recovery)
+	head, paused, err := loadSnapshot(diskdb, triedb, cache, root, recovery)
+	if paused {
+		log.Warn("Snapshot maintenance paused (syncing)")
+		return snap, nil
```

Comment:
Is it intentional to return `snap, nil` here ? `paused ` means it shouldn't be used, no?

Reply:
The point is to return a valid snap tree, but keep it empty. If it's empty, nothing can be retrieved or build on top, the only thing it can do it Rebuild, which is done on pivot commit (and which will reactivate it).